server/utils/message_parser_runner.ts

```diff
@@ -29,7 +33,17 @@ export class MessageParserRunner {
       } catch (e) {
         tempResult = [];
       }
-      results = [...results, ...tempResult];
+
+      results = [
+        ...results,
+        ...tempResult.map((item) => {
+          messageIndex++;
+          return {
+            ...item,
+            messageId: `${interaction.interaction_id}_${messageIndex}`,
```

Comment:
Shall we use `messageId: item.messageId || `${interaction.interaction_id}_${messageIndex}``? The messageId will be different after the parse order change.

Reply:
MessageId should only be generated by message_parser_runner so for the original parsed messaged item, there is no messageId field and for each parser, it is not recommend to add messageId into their messages response because they do not have enough context to generate a proper id.

Reply:
> after the parse order change.

`messageId` is just a runtime field used to patch and identify each message, not a field that will be persistent inside OpenSearch index.